salvo/src/lib/benchmark/base_benchmark.py

```diff
@@ -105,40 +159,8 @@ def run_image(self, image_name: str,
     """
     return self._docker_image.run_image(image_name, run_parameters)
 
-  def pull_images(self) -> List[str]:
-    """Retrieve all images necessary for the benchmark.
-
-    Retrieve the NightHawk and Envoy images defined in the control
-    object.
-
-    Returns:
-        a List of required image names that are retrievable. If any image is
-          unavailable, we return an empty list. The intent is for the caller to
-          build the necessary images.
-
-    Raises:
-        BenchmarkError: if a requested image is unavailable.
-    """
-    retrieved_images = []
-    images = self.get_images()
-
-    for image in [
-        images.nighthawk_benchmark_image,
-        images.nighthawk_binary_image,
-        images.envoy_image
-    ]:
-      # If the image name is not defined, we will have an empty string.
-      # For unit testing we'll keep this behavior. For true usage, we
-      # should raise an exception when the benchmark class performs its
-      # validation
-      if image:
-        retrieved_image = self._docker_image.pull_image(image)
-        log.debug(f"Retrieved image: {retrieved_image} for {image}")
-        if retrieved_image is None:
-          raise BenchmarkError("Unable to retrieve image: %s" % image)
-        retrieved_images.append(retrieved_image)
-
-    return retrieved_images
+  def execute_benchmark(self) -> None:
```

Comment:
(optional) A more idiomatic way of indicating that a method needs implementation on a derived class wold be to mark it as an abstract method:

https://docs.python.org/3/library/abc.html#abc.abstractmethod

Reply:
Right, I forgot about that.  I will update.